workers/unity/Packages/io.improbable.gdk.buildsystem/WorkerBuilder.cs

```diff
@@ -273,74 +239,78 @@ public static void Clean()
+                PlayerSettings.SetScriptingBackend(buildTargetGroup, buildContext.ScriptingImplementation);
+            }
+
+            try
             {
-                if (buildTarget == BuildTarget.Android && string.IsNullOrEmpty(EditorPrefs.GetString("AndroidSdkRoot")))
+                currentContext = buildContext;
+                var result = BuildPipeline.BuildPlayer(buildPlayerOptions);
+                if (result.summary.result != BuildResult.Succeeded)
                 {
-                    Debug.LogWarning($"Unable to build worker {workerType} for platform Android. " +
-                        $"Ensure you have the Android SDK set inside the Unity Editor Preferences.");
-                    return false;
-                }
+                    if (buildContext.BuildTargetConfig.Target == BuildTarget.Android &&
+                        string.IsNullOrEmpty(EditorPrefs.GetString("AndroidSdkRoot")))
+                    {
+                        Debug.LogWarning($"Unable to build worker {buildContext.WorkerType} for platform Android. " +
+                            $"Ensure you have the Android SDK set inside the Unity Editor Preferences.");
+                        return false;
+                    }
 
-                throw new BuildFailedException($"Build failed for {workerType}");
+                    throw new BuildFailedException($"Build failed for {buildContext.WorkerType}");
+                }
             }
-
-            if (buildTarget == BuildTarget.Android || buildTarget == BuildTarget.iOS)
+            catch (BuildFailedException)
             {
-                // Mobile clients can only be run locally, no need to package them
-                return true;
+                throw;
+            }
+            catch (Exception e)
+            {
+                throw new BuildFailedException(e);
+            }
+            finally
+            {
+                currentContext = null;
+                PlayerSettings.SetScriptingBackend(buildTargetGroup, activeScriptingBackend);
             }
 
-            if (buildTarget == BuildTarget.StandaloneOSX)
+            if (buildContext.BuildTargetConfig.Target == BuildTarget.StandaloneOSX)
             {
                 // Unity executable name has changed on MacOS
                 // This is a temp work around our Launcher trying to run the wrong file.
                 // See [UTY-2294]
                 CreateLaunchJson(workerBuildData.PackageName, Application.productName);
             }
 
-            var zipPath = Path.Combine(PlayerBuildDirectory, workerBuildData.PackageName);
-            var basePath = Path.Combine(Common.BuildScratchDirectory, workerBuildData.PackageName);
-            Zip(zipPath, basePath, targetEnvironment == BuildEnvironment.Cloud);
+            // Package up standalone builds
+            if (buildContext.BuildTargetConfig.Target != BuildTarget.Android &&
+                buildContext.BuildTargetConfig.Target != BuildTarget.iOS)
```

Comment:
Nit could have a `IsMobile` method on `BuildContext`?

Reply:
I'd rather leave this for now. Its not that annoying and will change again as I want to move the zipping of the finished build elsewhere.